EleutherAI/gpt-neox
# Model sampling/text generation

Text generation is working with no output (unconditional) and with an input file. By "working" I mean it is loading a model, creating output and saving it to file. The output is currently utter gibberish as I'm testing it on basically a randomly initialised model.

Steps to test:

**Generate a model:**
1. first change the `save-interval` to 10 or something small in the model config.
2. Run a model (note this now saves the model config in the checkpoint directory so you can easily determine what the model data corresponds to):

`./deepy.py pretrain_gpt2.py -d configs small.yml eleutherai_cluster.yml` and wait for a checkpoint to save.

**To test `unconditional` text generation:**
1. Make sure `"text-gen-type": "unconditional"` is set in the `text_generation.yml` config
2. Run an text generation job loading the generated model:

`./deepy.py text_gen_gpt2.py  -d configs small.yml eleutherai_cluster.yml text_generation.yml`

3. Output will be found in `/mnt/ssd-cluster/output/text_generation.txt`

**To test `input-file` text generation:**
1. Make sure `"text-gen-type": "input-file"` is set in the `text_generation.yml` config
2. Create an new line delineated input file at `/mnt/ssd-cluster/output/sample_input.txt`
2. Run a text generation job:

`./deepy.py text_gen_gpt2.py  -d configs small.yml eleutherai_cluster.yml text_generation.yml`

3. Output will be found in `/mnt/ssd-cluster/output/sample_output.txt`

Todo:
- [X] Model loading
- [X] Unconditional output
- [X] Output from file
- [X] Save model config to checkpoint folder
- [x] Create example script in examples folder
